GeneratorInterface/RivetInterface/plugins/HTXSRivetProducer.cc

```diff
@@ -108,13 +103,14 @@ void HTXSRivetProducer::produce(edm::Event& iEvent, const edm::EventSetup&) {
         } else if (nBs == 2 && nHs == 1 && nZs == 0) {
           m_HiggsProdMode = HTXS::BBH;
         }
-
-        _HTXS->setHiggsProdMode(m_HiggsProdMode);
       }
     }
 
-    if (_isFirstEvent) {
-      _analysisHandler.addAnalysis(_HTXS);
+    if (!_HTXS || !_HTXS->hasProjection("FS")) {
+      delete _HTXS;
```

Comment:
This should not be deleted. It will be deleted when `_analysisHandler` is re-assigned